mx.py

```diff
@@ -3590,6 +3590,8 @@ def update(self, vcdir, rev=None, mayPull=False, clean=False, abortOnError=False
         """
         if rev and mayPull and not self.exists(vcdir, rev):
             self.pull(vcdir, rev=rev, update=False, abortOnError=abortOnError)
+            if not self.exists(vcdir, rev):
+                abort('Fetch of %s succeeded\nbut did not contain requested revision %s.\nCheck that your remotes agree with the suite.py reposity location' % (vcdir, rev))
```

Comment:
can't we just `git fetch` from the URLs in suite.py instead?

btw: s/reposity/repository/